spec/ics-007-tendermint-client/README.md

```diff
@@ -200,18 +249,48 @@ function checkMisbehaviourAndUpdateState(
 }
 ```
 
+### Upgrades
+
+The chain which this light client is tracking can elect to write a special pre-determined key in state to allow the light client to update its client state (e.g. with a new chain ID or epoch) in preparation for an upgrade.
+
+As the client state change will be performed immediately, once the new client state information is written to the predetermined key, the client will no longer be able to follow blocks on the old chain, so it must upgrade promptly.
+
+```typescript
+function upgradeClientState(
+  clientState: ClientState,
+  newClientState: ClientState,
+  height: Height,
+  proof: CommitmentPrefix) {
+    // check proof of updated client state in state at predetermined commitment prefix and key
+    path = applyPrefix(clientState.upgradeCommitmentPrefix, clientState.upgradeKey)
+     // check that the client is at a sufficient height
+    assert(clientState.latestHeight >= height)
```

Comment:
Should we make this check more specific in implementation. (i.e. EpochNumber must be incremented)

Reply:
Yes, added that check, thanks.